arch/risc-v/src/litex/litex_perf.c

```diff
@@ -0,0 +1,83 @@
+/****************************************************************************
+ * arch/risc-v/src/litex/litex_perf.c
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this args for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/clock.h>
+#include <nuttx/config.h>
+
+#include <stdint.h>
+#include <time.h>
+
+#include "riscv_internal.h"
+
+#ifdef CONFIG_ARCH_PERF_EVENTS
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+#define NSEC_PER_CYCLE (NSEC_PER_SEC / CONFIG_LITEX_SYS_CORE_FREQ_HZ)
+#define CYCLE_PER_SEC  CONFIG_LITEX_SYS_CORE_FREQ_HZ
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: up_perf_init
+ ****************************************************************************/
+
+void up_perf_init(void *arg)
+{
+  (void)arg;
+}
+
+/****************************************************************************
+ * Name: up_perf_gettime
+ ****************************************************************************/
+
+unsigned long up_perf_gettime(void)
```

Comment:
but it's a general implementation, could be moved to the common place

Reply:
You're right, I'll move this to the common section and update as needed.